Integrate Custom Logger implementation and new Logger API. (#8199)
realm/realm-swift: CHANGELOG.md

x.y.z Release notes (yyyy-MM-dd)
=============================================================
### Enhancements

> * Added support for creating and add a custom logger, and implement your own logging logic.

Reviewer: ```suggestion
* Added support for creating a custom logger, to allow implementing your own logging logic.
```
